sdk/packages/core/src/api/validateOrder.test.ts

```diff
@@ -0,0 +1,107 @@
+import { testAccount, testOrder } from '@omni-network/test-utils'
+import { expect, test, vi } from 'vitest'
+import * as api from '../internal/api.js'
+import { validateOrder } from './validateOrder.js'
+
+// TODO calls as empty array should not be allowed // throw error
+
+test('default: native transfer order', async () => {
+  vi.spyOn(api, 'fetchJSON').mockResolvedValue({
+    accepted: true,
+  })
+
+  const resultPromise = validateOrder('http://localhost', {
+    owner: testAccount.address,
+    srcChainId: 1,
+    destChainId: 2,
+    calls: [
+      {
+        target: testAccount.address,
+        value: 0n,
+      },
+    ],
+    deposit: {
+      amount: 0n,
+    },
+    expense: {
+      amount: 0n,
+    },
+  })
+  await expect(resultPromise).resolves.toEqual({ accepted: true })
+  expect(api.fetchJSON).toHaveBeenCalledWith(
+    'http://localhost/check',
+    expect.any(Object),
+  )
+})
+
+test('default: order', async () => {
+  vi.spyOn(api, 'fetchJSON').mockResolvedValue({
+    accepted: true,
+  })
+  await expect(validateOrder('http://localhost', testOrder)).resolves.toEqual({
+    accepted: true,
+  })
+  expect(api.fetchJSON).toHaveBeenCalledWith(
+    'http://localhost/check',
+    expect.any(Object),
+  )
+})
+
+test('behaviour: resolves if response is supported error object', async () => {
+  const response = {
+    error: {
+      code: 1,
+      message: 'an error',
+    },
+  }
+  vi.spyOn(api, 'fetchJSON').mockResolvedValue(response)
+  await expect(validateOrder('http://localhost', testOrder)).resolves.toBe(
+    response,
+  )
+})
+
+test('behaviour: resolves if response is supported rejection object', async () => {
+  const response = {
+    rejected: true,
+    rejectReason: 'a reason',
+    rejectDescription: 'a description',
+  }
+  vi.spyOn(api, 'fetchJSON').mockResolvedValue(response)
+  await expect(validateOrder('http://localhost', testOrder)).resolves.toBe(
+    response,
+  )
+})
+
+test.each([
+  'test',
+  {},
+  { rejected: true },
+  { rejected: true, rejectReason: 'a reason' },
+  { rejecetd: true, rejectDescription: 'a description' },
+])('behaviour: throws if response is not valid: %s', async (mockReturn) => {
+  vi.spyOn(api, 'fetchJSON').mockResolvedValue(mockReturn)
+
+  const expectRejection = expect(async () => {
+    await validateOrder('http://localhost', testOrder)
+  }).rejects
+  await expectRejection.toBeInstanceOf(Error)
```

Comment:
do we want to test our specific error types?

Reply:
Yes eventually I think we should be able to differentiate between input errors (can't encode the order) and API errors or exceptions, I think we should consider it as part of a larger refactoring though.